lib/std/fs/path.zig

```diff
@@ -1604,6 +1610,23 @@ pub fn ComponentIterator(comptime path_type: PathType, comptime T: type) type {
                 .path = self.path[0..end_index],
             };
         }
+
+        /// Like `previous` but skips '.' entries and handles '..' entries
+        /// by skipping its parent directory.
+        pub fn previousResolveDots(self: *Self) ?Component {
+            var dot_dot_count: usize = 0;
+            while (true) {
+                const component = self.previous() orelse return null;
+                if (std.mem.eql(u8, component.name, ".")) {
+                    // ignore
+                } else if (std.mem.eql(u8, component.name, "..")) {
+                    dot_dot_count += 1;
+                } else {
+                    if (dot_dot_count == 0) return component;
+                    dot_dot_count -= 1;
+                }
+            }
+        }
```

Comment:
I haven't thought it through fully, but it seems like this function is a bit fraught due to `..` and symlink resolution, e.g.:

- https://github.com/ziglang/zig/issues/7751
- https://github.com/ziglang/zig/issues/4658

Reply:
I believe the only case that would be a concern is if an import string contains `..` components that are not at the beginning.  So long as they only appear at the beginning, it should behave the same on all platforms.  I'd advocate for a restriction on import paths that `..` components can only appear at the beginning.

Reply:
Sorry, I meant it is fraught as a public API of `ComponentIterator`, since it could lead to unexpected behavior with regards to symlinks. At the very least the doc comment should mention something about symlinks.

If this would work without problems for the `@import` use case, then moving this logic to the usage site(s) and removing the API from `ComponentIterator` might make sense.

Reply:
I see.  Since this function is used in both the compiler itself and the std library for a test I've opted to leave it where it is and add a clarifying comment.  Let me know if you have a better idea:
```zig
        /// For non-windows systems, this function does not resolve '..' components
        /// the same way the filesystem would. If a parent directory of '..' is a
        /// symlink to anything other then an immediate subdirectory then the resulting
        /// path would differ.
```

Reply:
Nevermind, I realized the test in `std` could just use `previous` instead.  I've moved it into `Zcu/PerThread.zig` to keep it private.